### PR TITLE
Add enum generation

### PR DESCRIPTION
**Pull request by fakenmc**
Merge enum generation originally in https://github.com/Venipa/csharpextensions/blob/master/src/extension.ts

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -122,7 +128,8 @@ function openTemplateAndSaveNewFile(type: string, namespace: string, filename: s
         .then(doc => {
             let text = doc.getText()
                 .replace('${namespace}', namespace)
-                .replace('${classname}', filename);
+                .replace('${classname}', filename)
+                .replace('${enumname}', filename);
```
Comment (KreativJos) [inline src/extension.ts:132]: Why use 'enumname' here, when in both the other template files, 'classname' is used?
I like consistency. Why not use 'classname', or change the interface one to 'interfacename', or change all to something like 'itemname' or just 'name'?

Comment (fakenmc) [reply src/extension.ts]: You're right. I've more or less auto-merged the commits by the original author, Venipa, and I missed that when fixing the merge conflicts. I've pushed a more consistent version. Thanks.

File: src/extension.ts [hunk 2]
```diff
@@ -31,9 +32,14 @@ function createInterface(args: any) {
     promptAndSave(args, 'interface');
 }
 
-function promptAndSave(args: any, templatetype: string) {
-    if (args == null) args = { _fsPath: vscode.workspace.rootPath }
+function createEnum(args) {
```
Comment (KreativJos) [inline src/extension.ts:35]: The compiler has issues with missing the type here.
```suggestion
function createEnum(args: any) {
```

File: src/extension.ts [hunk 3]
```diff
@@ -31,9 +32,14 @@ function createInterface(args: any) {
...
+    promptAndSave(args, 'enum');
+}
 
+function promptAndSave(args, templatetype: string) {
```
Comment (KreativJos) [inline src/extension.ts:39]: And here
```suggestion
function promptAndSave(args: any, templatetype: string) {
```

**Comment by fakenmc** [comment]
Thanks!

**Comment by KreativJos** [comment]
The new version is in the verifying state. It will show up in a few minutes, or at least in an hour.
Thank you for contributing!

**Pull request merged**